src/transformers/trainer_pt_utils.py

```diff
@@ -42,17 +42,50 @@
 logger = logging.get_logger(__name__)
 
 
-def nested_concat(tensors, new_tensors, dim=0):
-    "Concat the `new_tensors` to `tensors` on `dim`. Works for tensors or nested list/tuples of tensors."
+def torch_pad_and_concatenate(tensor1, tensor2, pad_idx=-100):
```

Comment:
Can the argument be `padding_id`, as that's the wording used elsewhere? (this applies to all methods used here)